File: services/flink_service.py
from dataclasses import dataclass
from typing import *

from pyflink.table import TableEnvironment, EnvironmentSettings, StreamTableEnvironment
from pyflink.datastream import StreamExecutionEnvironment
from pyflink.common import Configuration


@dataclass
class FlinkService:
    """
        jars_path: full path to jar files
        settings: a dicts of flink's settings
    """
    jars_path: List[str]
    settings: dict

    def __post_init__(self):
        config = Configuration()
        for k, v in self.settings.items():
            config.set_string(k, v)
        env_settings = EnvironmentSettings.in_streaming_mode()
        self.env =StreamExecutionEnvironment.get_execution_environment()
        self.env.configure(config)

        self.t_env = StreamTableEnvironment.create(environment_settings=env_settings)
        # self.t_env.get_config().add_configuration(config)

File: pipeline/app/src/main/java/org/example/Service/FlinkService.java
package org.example.Service;

import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.table.api.EnvironmentSettings;
import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;

import java.util.Map;

public class FlinkService {
    private final Configuration config;
    private StreamTableEnvironment tEnv;

    public FlinkService() {
        this.config = new Configuration();
    }

    public void setString(Map<String, String> configMap){
        for(Map.Entry<String,String> entry : configMap.entrySet()){
            this.config.setString(entry.getKey(), entry.getValue());
        }
    }

    public void createTableEnv(){
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(this.config);
        env.enableCheckpointing(5000);
        EnvironmentSettings settings = EnvironmentSettings.inStreamingMode();
        this.tEnv = StreamTableEnvironment.create(env, settings);
    }

    public StreamTableEnvironment getTEnv(){
        return this.tEnv;
    }
}
